tests/benchmarks/multicore/idle_ppr/src/main.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(idle_ppr, LOG_LEVEL_INF);
+
+/* Variables used to make CPU active for ~1 second */
+static struct k_timer my_timer;
+static bool timer_expired;
+
+void my_timer_handler(struct k_timer *dummy)
+{
+	timer_expired = true;
+}
+
+int main(void)
+{
+	int counter = 0;
+
+	LOG_INF("Multicore idle_ppr test on %s", CONFIG_BOARD_TARGET);
+	LOG_INF("Main sleeps for %d ms", CONFIG_TEST_SLEEP_DURATION_MS);
+
+	k_timer_init(&my_timer, my_timer_handler, NULL);
+
+	/* Run test forever */
+	while (1) {
+		timer_expired = false;
+
+		/* start a one-shot timer that expires after 1 second */
+		k_timer_start(&my_timer, K_MSEC(1000), K_NO_WAIT);
+
+		/* Keep CPU active for ~ 1 second */
+		while (!timer_expired) {
+			k_busy_wait(10000);
+			k_yield();
+		}
+
+		LOG_INF("Run %d", counter);
```

Comment:
Maybe GPIO blink to confirm that cores are up and running in the S2RAM/IDLE variant (when console is disabled).

Reply:
Or maybe some busy_wait to detect that.

Reply:
busy_wait() is in line 38.
All three cores (APP, Radio, PPR) execute the same code.
I'm afraid of situation when APP and Radio is busy_waiting while PPR is dead -> we may not detect corrupted state of PPR.

Reply:
In IDLE/S2RAM console is disabled -> no way CI will confirm that core is executing. I add LED blinking for each core. At least manual test will confirm if core wakes up from sleep state.